### PR TITLE
fix(vsts): ado project limit

### PR DESCRIPTION
**Pull request by scefali**
Some organizations have more than 100 ADO projects which is a problem because we only fetch 100 at a time. This PR implements pagination by providing a `$skip` parameter and incrementing it by the page size (`$top`). I have chosen this approach over using a continuation token for a few reasons:

1. The API version we currently use (4.1) does not include a continuation token in the response
2. The continuation token comes in a response header which we normally don't have access to (we'd have to refactor the base integration client to return more than just the body of the response)

See docs: https://docs.microsoft.com/en-us/rest/api/azure/devops/core/projects/list?view=azure-devops-rest-6.1

### CODE REVIEW

File: src/sentry/integrations/vsts/client.py [hunk 1]
```diff
@@ -213,11 +213,25 @@ def get_project(self, instance, project_id):
         )
 
     def get_projects(self, instance):
-        # TODO(dcramer): VSTS doesn't provide a way to search, so we're
-        # making the assumption that a user has 100 or less projects today.
-        return self.get(
-            VstsApiPath.projects.format(instance=instance), params={"stateFilter": "WellFormed"}
-        )
+        limit = 100  # 100 is max
```
Comment (MeredithAnya) [inline src/sentry/integrations/vsts/client.py:216]: I thought I saw somewhere on the docs that 200 was the max value for `$top` but now I can't seem to find it, but ill try looking again.

File: src/sentry/integrations/vsts/client.py [hunk 2]
```diff
@@ -213,11 +213,25 @@ def get_project(self, instance, project_id):
...
+        offset = 0
+        projects = []
+
+        # no one should have more than 1000 projects
```
Comment (MeredithAnya) [inline src/sentry/integrations/vsts/client.py:220]: Im actually wondering if 1000 will actually be a possibility

**Review by MeredithAnya — APPROVED** [review]
I still don't think 1000 is actually too crazy but this would be an easy thing to update if that becomes a problem

**Comment by scefali** [comment]
> I still don't think 1000 is actually too crazy but this would be an easy thing to update if that becomes a problem

@MeredithAnya 1000 is 4x what a certain company has right now. I agree, we can update it later if it becomes a problem which seems unlikely right now.

**Comment by joshuarli** [comment]
So this broke py3, but TIL to make them required it's a repository setting, and so the ability to merge here was not blocked.

**Pull request merged**